src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -38,42 +38,67 @@ public class AddCommandParser implements Parser<AddCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public AddCommand parse(String args) throws ParseException {
-        ArgumentMultimap argMultimap =
+        ArgumentMultimap argMultimap = processRawCommand(args);
+        Person person = createPerson(argMultimap);
+        return new AddCommand(person);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
+    /**
+     * Processes the given {@code String} into an ArgumentMultimap in the context of an AddCommand.
+     *
+     * @param args Raw command string.
+     * @return ArgumentMultimap containing argument values to create a Person with.
+     * @throws ParseException if the string contains invalid arguments or duplicate arguments for Person fields that
+     *      require exactly one argument.
+     */
+    private ArgumentMultimap processRawCommand(String args) throws ParseException {
+        ArgumentMultimap argumentMultimap =
```

Comment:
Would it be better if we create a new class with static methods to do this? Else, we might have to add this method into every single command we have parser.

Reply:
There are certain restrictions on the input that this method checks for e.g. duplicates, invalid preambles that are different for each command. Also, not all commands need an argument multimap e.g. help, list.